src/ripple/app/consensus/RCLConsensus.cpp

```diff
@@ -633,29 +633,31 @@ RCLConsensus::Adaptor::doAccept(
         JLOG(j_.info()) << "We closed at "
                         << closeTime.time_since_epoch().count();
         using usec64_t = std::chrono::duration<std::uint64_t>;
-        usec64_t closeTotal =
-            std::chrono::duration_cast<usec64_t>(closeTime.time_since_epoch());
+        std::vector<usec64_t> closeTimes;
         int closeCount = 1;
 
         for (auto const& p : rawCloseTimes.peers)
         {
-            // FIXME: Use median, not average
             JLOG(j_.info())
                 << std::to_string(p.second) << " time votes for "
                 << std::to_string(p.first.time_since_epoch().count());
+
             closeCount += p.second;
-            closeTotal += std::chrono::duration_cast<usec64_t>(
-                              p.first.time_since_epoch()) *
-                p.second;
+
+            auto time = std::chrono::duration_cast<usec64_t>(p.first.time_since_epoch());
+            for(int pi = 0; pi < p.second; ++pi)
+              closeTimes.push_back(time);
```

Comment:
Potential for improving readability/comprehension by replacing this loop with:

    closeTimes.insert(closeTimes.end(), p.second, p.first.time_since_epoch());